File: cpp/234.cpp
#include <vector>
using namespace std;

struct ListNode {
  int val;
  ListNode *next;
  ListNode(int x) : val(x), next(NULL) {}
};

class Solution {
public:
  bool isPalindrome(ListNode *head) {
    vector<int> v;
    while (head) {
      v.push_back(head->val);
      head = head->next;
    }
    int n = v.size();
    for (int i = 0; i < n / 2; i++)
      if(v[i] != v[n-1-i])
        return false;
    return true;
  }
};

File: python/234.py
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def isPalindrome(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        l = []
        while head:
            l.append(head.val)
            head = head.next
        return l == l[::-1]
